staging/src/k8s.io/apiserver/pkg/util/flowcontrol/rule.go

```diff
@@ -0,0 +1,109 @@
+package filters
+
+import (
+	rmtypesv1alpha1 "k8s.io/api/flowcontrol/v1alpha1"
+	"k8s.io/apiserver/pkg/authentication/serviceaccount"
+)
+
+func matchesFlowSchema(digest RequestDigest, flowSchema *rmtypesv1alpha1.FlowSchema) bool {
+	for _, policyRule := range flowSchema.Spec.Rules {
+		subjectMatches := false
+		for _, subject := range policyRule.Subjects {
+			if matchesSubject(digest, subject) {
+				subjectMatches = true
+			}
+		}
+		if !subjectMatches {
+			return false
+		}
+
+		policyRuleMatches := !matchesPolicyRule(digest, policyRule.Rule)
+
+		if subjectMatches && policyRuleMatches {
+			return true
+		}
+	}
+	return false
+}
+
+func matchesSubject(digest RequestDigest, subject rmtypesv1alpha1.Subject) bool {
+	user := digest.User
+	requestingNamespace := digest.RequestInfo.Namespace
```

Comment:
This is the namespace of the object that the request is requesting to operate on, NOT the namespace of the user making the request.

Reply:
removed